apps/src/block_utils.js

```diff
@@ -429,6 +429,30 @@ const DUMMY_INPUT = 'dummy';
 const STATEMENT_INPUT = 'statement';
 const FIELD_INPUT = 'field';
 
+/**
+ * Splits a blockText into labelled inputs, each match will a label followed by
+ * an input. The label is an arbitrary (possibly empty) string. The input is
+ * either a real named input like '{VALUE}', a newline, or if it matched a
+ * trailing label, nothing.
+ */
+const LABELED_INPUTS_REGEX = /.*?({[^}]*}|\n|$)/gm;
+
+/**
+ * Splits a labeled input into its parts. The resulting groups are:
+ * 1: the label
+ * 2: the input (a named input like '{VALUE}', a newline, or nothing)
+ * 3: the input's name (e.g. 'VALUE')
+ */
+const LABELED_INPUT_PARTS_REGEX = /(.*?)({([^}]*)}|\n|$)/m;
+
+const findInputConfig = (args, inputName) => {
```

Comment:
Maybe clearer as `findAndRemoveInputConfig`?  Not clear initially that this method mutates the `args` array.

Please add a JSDoc.